packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,543 @@
+  console.log(selectedItems);
+};
+
+const onSortChange = ({ headerKey, sortType }) => {
+  console.log({headerKey, sortType});
+};
+```
+
+
+```jsx
+// Composable Table
+
+<Table 
+    data={data} // This contains the actual data to be rendered in the table which would be retrieved from some API
+    selectionType='single' 
+    onSelectionChange={onSelectionChange} 
+    sortFunctions={sortFunctions} 
+    onSortChange={onSortChange}
+    isLoading={false}
+    cellDensity='normal'
+    showZebraStripes={true}
+    pagination={{
+      limit: 10,
+      offset: 0,
+    }}
+>
+{(tableData) => (
+  <>
+    <TableToolbar title="Users" itemsSelectedTitle={`${selectedUsersCount} Users selected`}>
+        <ToolbarActions>
+            <Button>Export</Button>
+        </ToolbarActions>
+    </TableToolbar>
+
+    <TableHeader isSticky={true}>
+      <TableHeaderRow>
+         <TableHeaderCell headerKey='firstName'>
+          First Name
+        </TableHeaderCell>
+        <TableHeaderCell headerKey='lastName'>
+          Last Name
+        </TableHeaderCell>
+        <TableHeaderCell headerKey='balance'>
+          Balance
+          <Tooltip content='Current bank balance'>
+            <InfoIcon size='medium' />
+          </Tooltip>
+        </TableHeaderCell>
+       </TableHeaderRow>
+     </TableHeader>
+
+    <TableBody>
+      {tableData.map((row) => (
+        <TableRow onClick={console.log} isDisabled={false}>
+          <TableCell>
+            {row.firstName}
+          </TableCell>
+          <TableCell>
+            {row.lastName}
+          </TableCell>
+          <TableCell>
+            {row.balance}
+            <Badge>Low</Badge>
+          </TableCell>
+        <TableRow>
+      )}
+    </TableBody>
+
+    <TableFooter>
+      <TableFooterRow>
+        <TableFooterCell columnStart='1' columnEnd='4'>
+          Total
+        </TableFooterCell>
+        <TableFooterCell>
+          {computedTotalBalance}
+        </TableFooterCell>
+      </TableFooterRow>
+    </TableFooter>
+
+    <TablePagination 
+      navigationType='compact' 
+      label='1-10 of 100'
+      labelPosition='left'
+      showRowCountPicker
+      rowCount='25'
+      onPaginationChange={console.log}
+      onRowCountPickerChange={console.log}
+      /> 
+   </>
+  )
+}
+</Table>
+```
+
+##### Pros
+- Composable API is more intuitive and resembles the native HTML structure of a table
+- It is easier to understand since the API design is similar to rest of the Blade components
+- It is easier to extend and add new feature to individual table components
+- We could leverage tree-shaking to only import the components that are being used in the consumer's table
+
+##### Cons
+- Composable API is more verbose and requires more effort to create a table by composing together multiple components
+- Most popular table libraries like AG Grid, React Table, etc. use a compact API which is battle tested and developers are more familiar with the same
```

Comment:
React table is headless table library, You have to compose your own table using the data provided from React table. Not sure this falls into Compact API.

Reply:
https://codesandbox.io/p/sandbox/github/tanstack/table/tree/main/examples/react/column-groups?embed=1&file=%2Fsrc%2Fmain.tsx%3A1%2C1 Can we replicate this example?

Reply:
It sort of does both compact and composable. It uses the ideas of compact API to create column objects and then the consumers have to map over it and compose their tables. Its not _truly_ composable in a sense that it doesn't expose composable components hence I categorized it as compact API